### ISSUE TITLE
Target size (torch.Size([1, 3, 512, 512])) must be the same as input size (torch.Size([1, 1, 512, 512]))

### ISSUE THREAD

**Issue by DanteDream** [milesial/Pytorch-UNet#149]
IMy classification is dichotomous, but I have encountered this problem.

**Comment by milesial** [comment]
You are loading your true masks as 3 channels, you should have only one channels if you have 2 classes

**Comment by laclouis5** [comment]
Should the masks be 1 channel too for 1 class? i.e a 8 bit PNG image with 1's and 0's?

I tried to train this networks but the loss stays very stable and close to zero, and the dice coefficient for the validation set oscillates but is always a fraction of the validation test size (i.e my dataset is 16 images, dice loss is `dc = n / 16` with `n` in 0...16).

**Comment by milesial** [comment]
Yes 1 channel for 1 class sounds fine. If you think about it, 1 class and 2 classes (1class + background) are the same.

Your dice problem seems strange, it seems that in your case a validation image is either all right (1) or all wrong (0). It might be a problem of loading your true mask in the wrong format, so the dice computation is confused. I would debug and set breakpoints just before the dice computation and inspect your predicted and true masks.
